opensrp-tasking/src/test/java/org/smartregister/tasking/presenter/TaskingHomePresenterTest.java

```diff
@@ -0,0 +1,945 @@
+        assertEquals(IN_PROGRESS.name(), featureCollectionArgumentCaptor.getValue().features().get(0).getStringProperty(TASK_STATUS));
+
+        verify(taskingHomeInteractor).fetchInterventionDetails(BLOOD_SCREENING, structureId, false);
+    }
+
+    @Test
+    public void testResetFeatureTasks() throws Exception {
+        String structureId = "id1";
+        com.cocoahero.android.geojson.Feature feature1 = new com.cocoahero.android.geojson.Feature();
+        feature1.setIdentifier(structureId);
+        feature1.setProperties(new JSONObject());
+
+
+        Feature mapboxFeature = Feature.fromJson(feature1.toJSON().toString());
+        assertNull(mapboxFeature.getStringProperty(TASK_BUSINESS_STATUS));
+        assertNull(mapboxFeature.getStringProperty(TASK_STATUS));
+        assertNull(mapboxFeature.getStringProperty(TASK_IDENTIFIER));
+        assertNull(mapboxFeature.getStringProperty(TASK_CODE));
+        assertNull(mapboxFeature.getStringProperty(FEATURE_SELECT_TASK_BUSINESS_STATUS));
+
+        Whitebox.setInternalState(presenter, "featureCollection",
+                FeatureCollection.fromFeatures(new Feature[]{mapboxFeature}));
+
+        Task task = TestingUtils.getTask("entityid");
+
+        presenter.resetFeatureTasks(structureId, task);
+
+        verify(view).setGeoJsonSource(featureCollectionArgumentCaptor.capture(), any(), booleanArgumentCaptor.capture());
+        assertFalse(booleanArgumentCaptor.getValue());
+        assertEquals(task.getBusinessStatus(), featureCollectionArgumentCaptor.getValue().features().get(0).getStringProperty(TASK_BUSINESS_STATUS));
+        assertEquals(task.getBusinessStatus(), featureCollectionArgumentCaptor.getValue().features().get(0).getStringProperty(FEATURE_SELECT_TASK_BUSINESS_STATUS));
+        assertEquals(task.getStatus().name(), featureCollectionArgumentCaptor.getValue().features().get(0).getStringProperty(TASK_STATUS));
+        assertEquals(task.getIdentifier(), featureCollectionArgumentCaptor.getValue().features().get(0).getStringProperty(TASK_IDENTIFIER));
+        assertEquals(task.getCode(), featureCollectionArgumentCaptor.getValue().features().get(0).getStringProperty(TASK_CODE));
+
+    }
+
+    @Test
+    public void testOnFormSaveFailureForSprayEvent() {
+
+        presenter.onFormSaveFailure(SPRAY_EVENT);
+
+        verify(view).hideProgressDialog();
+        verify(view).displayNotification(R.string.form_save_failure_title, R.string.spray_form_save_failure);
+    }
+
+    @Test
+    public void testOnFormSaveFailureForStructureEvent() {
+
+        presenter.onFormSaveFailure(REGISTER_STRUCTURE_EVENT);
+
+        verify(view).hideProgressDialog();
+        verify(view).displayNotification(R.string.form_save_failure_title, R.string.add_structure_form_save_failure);
+    }
+
+//    @Test
+//    public void testOnInterventionFormDetailsFetched() {
```

Comment:
I am not sure of the intention when commenting these out. Could you do something about it? Create an issue, todo, remove, fix